<launch>
	<arg name="image_topic" default="/kitti/camera_color_right/image_raw" />
	<arg name="mmdet_pub_rate" default="50" />
	<arg name="is_service" default="false" />
	<arg name="visualization" default="true" />

	<arg name="config_path" default="$(find mmsegmentation_ros)/mmsegmentation/configs/pspnet/pspnet_r50-d8_512x1024_40k_cityscapes.py" />
	<arg name="checkpoint_path" default="$(find mmsegmentation_ros)/mmsegmentation/workdir/latest.pth" />

	<node pkg= "mmsegmentation_ros" name="mmsegmentor" type="mmsegmentor.py" output="screen" respawn="true">
		<remap from="~image_topic" to="$(arg image_topic)"/>
		<param name="~publish_rate" value="$(arg mmdet_pub_rate)" />
		<param name="~is_service" value="$(arg is_service)" />
		<param name="~visualization" value="$(arg visualization)" />

		<param name="~config_path" value="$(arg config_path)" />
		<param name="~checkpoint_path" value="$(arg checkpoint_path)" />
	</node>
</launch>
